x-pack/plugins/infra/server/lib/alerting/metric_threshold/metric_threshold_executor.ts

```diff
@@ -246,6 +246,22 @@ export const createMetricThresholdExecutor = (libs: InfraBackendLibs) =>
         });
       }
     }
+
+    const { getRecoveredAlerts } = services.alertFactory.done();
+    const recoveredAlerts = getRecoveredAlerts();
+    for (const alert of recoveredAlerts) {
+      const recoveredAlertId = alert.getId();
+      const viewInAppUrl = getViewInAppUrl(libs.basePath, LINK_TO_METRICS_EXPLORER);
+      const context = {
+        group: recoveredAlertId,
```

Comment:
Why don't we add `alertState: stateToAlertMessage[nextState],` to the context for recovered alerts? Similarly we could set reason and value. Is there a way you didn't include these in the context?

Reply:
Added `alertState` to context

Reply:
The `reason` message is ignored by the alert system and the `value` would require us to load ALL the data instead of just the "entities" that matched negating the optimizations we gained from pushing the evaluations to ES.